webapp/src/pages/offer/components/OfferComponent.js

```diff
@@ -0,0 +1,214 @@
+import React, {Component} from "react";
+import {withRouter} from "react-router";
+import TextArea from "react";
+import {Chart} from "react-google-charts";
+import { userService } from "../../../services/user";
+import { offerServices  } from "../../../services/offer/offer.service";
+import { profileService} from "../../../services/profilelist/profilelist.service";
+
+const DEFAULT_EVENT_ID = process.env.REACT_APP_DEFAULT_EVENT_ID || 1;
+
+class Offer extends Component {
+  
+  constructor(props) {
+        super(props);
+  
+        this.state = {
+          user:{user_id:0},
+          userProfile : [],
+          loading: true,
+          error: "",
+          offer_id: null,
+          rejected_reason:"",
+          rejected:false,
+          showReasonBox:false,
+          accepted: false,
+          offerList: [],
+          category:""
+        };
+      }
+    
+      
+      handleChange = field => {
+        return event => {
+          this.setState({
+            [field.name]: event.target.value
+          });
+        };
+      };
+
+      buttonSubmit(){
+         const {offer_id,accepted,rejected,rejected_reason} = this.state;
+   
+        offerServices.updateOffer(offer_id, DEFAULT_EVENT_ID, accepted, rejected, rejected_reason)
+        .then(response=>{
+          if (response.msg === "succeeded") {
+            this.setState({
+              addedSucess: true,
+              notFound: false
+            });
+          } else if (response.error) {
+            this.setState({
```

Comment:
you need do display the error here. But get the functional part right. I'll fix this tomorrow morning

Reply:
SInce you seem to have fixed the other things, can we sort this handling of the error properly now (as you've done with the others :)